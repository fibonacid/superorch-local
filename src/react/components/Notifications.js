import React from 'react';
import PropTypes from 'prop-types';
import FlashMessage from "./FlashMessage";

function Notifications(props) {
  return (
    <ul data-test={'NotificationsComponent'}>
      {props.notifications.map((notification, i) => (
        <FlashMessage key={i}>
          {notification.message}
        </FlashMessage>
      ))}
    </ul>
  )
}

Notifications.propTypes = {
  notifications: PropTypes.arrayOf(PropTypes.object)
};

export default Notifications;
