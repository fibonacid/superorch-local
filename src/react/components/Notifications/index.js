import {connect} from "react-redux";
import Notifications from "./Notifications";

const mapStateToProps = state => ({
  notifications: state.flash
});

export default connect(mapStateToProps)(Notifications)
