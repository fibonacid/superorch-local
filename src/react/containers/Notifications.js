import {connect} from "react-redux";
import Notifications from "../components/Notifications";

const mapStateToProps = state => ({
  notifications: state.flash
});

export default connect(mapStateToProps)(Notifications)
