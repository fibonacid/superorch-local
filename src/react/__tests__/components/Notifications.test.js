import React from "react";
import {shallow} from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from "../../store";
import Notifications from "../../components/Notifications";

const mockStore = configureStore();

describe("Notifications", () => {
  it ('should render my component', () => {
    // mount component (shallow)
    const component = shallow(
      <Provider store={mockStore}>
        <Notifications/>
      </Provider>
    );
    // unmount component
    component.unmount();
  });
});
