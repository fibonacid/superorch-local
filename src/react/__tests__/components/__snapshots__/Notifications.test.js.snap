// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Component When has props Should renderer without errors 1`] = `
<Notificationsstyles__List
  data-test="NotificationsComponent"
>
  <Notificationsstyles__FlashMessage
    key="0"
  >
    First Message
  </Notificationsstyles__FlashMessage>
  <Notificationsstyles__FlashMessage
    key="1"
  >
    Second Message
  </Notificationsstyles__FlashMessage>
</Notificationsstyles__List>
`;
